opsdroid/database/sqlite/__init__.py

```diff
@@ -116,3 +116,18 @@ def __init__(self, config, opsdroid=None):
 
         self.client = _db
         return data
+
+    async def delete(self, key):
+        """Delete data from the database for a given key.
+
+        Args:
+            key (string): The key to delete in the database.
+
+        """
+        _LOGGER.debug(_("Deleting %s from sqlite"), key)
+
+        async with aiosqlite.connect(self.db_file, **self.conn_args) as _db:
+            cur = await _db.cursor()
+            await cur.execute("DELETE FROM {} WHERE key=?".format(self.table), (key,))
+
+        self.client = _db
```

Comment:
This assignment seems strange. But it seems to be a common pattern throughout this database module.

I expect we should be creating `self.client` in the connect method, and probably not in a context manager either. We should then be closing the client in a disconnect method (which doesn't appear to be implemented here). Then the get, put, delete should all use the client instead of making a new connection.

This feels out of scope for this PR. If you want to fix it now then please go ahead, but I'm also happy to just take it as is. Up to you.

Reply:
Made the changes and added a `disconnect` method too.